### ISSUE TITLE
Make adding semicolon configurable for "Extract function"

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#18780]
_From @davedmarketing on September 24, 2017 17:53_

- VSCode Version: (1.16.1)
- OS Version: OSX 10.11.6

Steps to Reproduce: 

When I use "Extract function" it automatically adds ";". I don't use ";" when writing javascript except for inline "line breaks" or in front of self-executing functions so this causes a discrepancy.

Reproduces without extensions: Yes/No

_Copied from original issue: Microsoft/vscode#34911_

**Comment by mjbvz** [comment]
May fall within scope of #19882

**Comment by mjbvz** [comment]
Just a heads up @andrewbranch, I did a quick test on this issue it looks like this was not fixed by your work on #19882

Here's a quick repo that triggers the detector not to use semicolon for auto imports:

```js
console.log(1)
console.log(2)
console.log(3)
console.log(4)
console.log(5)
console.log(6)

console.log(1 + 2) // extract `1 + 2` here
```
